### PR TITLE
Document (Group)Cog interaction_check

### PR DESCRIPTION
**Pull request by Soheab**
## Summary

A couple of lines that were left over from when `app_commands.Group` and `commands.Cog` could be subclassed call a method named `interaction_check` on the binding but this was never documented. Since there is an intense desire for a cog-level app commands check, I have defined such special method in `commands.Cog` and documented it on `GroupCog`.

This was discussed in following bikeshedding thread: ["Cog.cog_check for app commands"](https://canary.discord.com/channels/336642139381301249/1060563548138381333) in the server.

## Checklist

- [x] If code changes were made then they have been tested.
    - [x] I have updated the documentation to reflect the changes.
- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: discord/ext/commands/cog.py [hunk 1]
```diff
@@ -585,6 +585,16 @@ def cog_check(self, ctx: Context[BotT]) -> bool:
         """
         return True
 
+    @_cog_special_method
+    def interaction_check(self, interaction: discord.Interaction[BotT], /) -> bool:
+        """A special method that registers as a :func:`discord.app_commands.check`
+        for every app command and subcommand in this cog.
+
+        This function **can** be a coroutine and must take a sole parameter,
+        ``interaction``, to represent the :class:`~discord.Interaction`.
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:594]: Needs a versionadded, which is actually 2.0 and not 2.2.

File: discord/ext/commands/cog.py [hunk 2]
```diff
@@ -585,6 +585,16 @@ def cog_check(self, ctx: Context[BotT]) -> bool:
         """
         return True
 
+    @_cog_special_method
+    def interaction_check(self, interaction: discord.Interaction[BotT], /) -> bool:
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:589]: This has to be async.

File: discord/ext/commands/cog.py [hunk 3]
```diff
@@ -585,6 +585,16 @@ def cog_check(self, ctx: Context[BotT]) -> bool:
...
+        """A special method that registers as a :func:`discord.app_commands.check`
+        for every app command and subcommand in this cog.
+
+        This function **can** be a coroutine and must take a sole parameter,
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:593]:
```suggestion
        This function **must** be a coroutine and must take a sole parameter,
```

**Review by Rapptz — CHANGES_REQUESTED** [review]

File: discord/ext/commands/cog.py (hunk 2 above)
Comment (Soheab) [reply discord/ext/commands/cog.py:589]: But the library uses utils.maybe_coroutine? 
https://github.com/Rapptz/discord.py/blob/f7e97954950ffb0e34238d70813454caa6f1a3ae/discord/app_commands/commands.py#L991-L995

File: discord/ext/commands/cog.py [hunk 4]
```diff
@@ -585,6 +585,18 @@ def cog_check(self, ctx: Context[BotT]) -> bool:
         """
         return True
 
+    @_cog_special_method
+    def interaction_check(self, interaction: discord.Interaction[BotT], /) -> bool:
```
Comment (Rapptz) [inline discord/ext/commands/cog.py:589]: This PR is failing because `BotT` doesn't quite match `ClientT` so you'll have to use the latter.

**Review by Rapptz — APPROVED** [review]

**Pull request merged**